### PR TITLE
Scope Handling Improvements + Tests

### PR DESCRIPTION
**Pull request by ikreymer**
Ensure custom scope includes take precedence over non-custom scope
Set scope to 'custom' if include/scopeIncludeRx is provided.
Use `scopeType` instead of `type` for scope at both top and seed level.
Add --scopeIncludeRx and --scopeExcludeRx for more specific cmdline options, still allow --include/--exclude
Add tests for various top-level and seed-level scope configurations
Update includes to work with tests
Tests: switch to yarn instead of npm for consistency, fix test issues
Update README
Bump to 0.4.0-beta.3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,7 +124,7 @@ Options:
       --userAgentSuffix                     Append suffix to existing browser
                                             user-agent (ex: +MyCrawler,
                                             info@example.com)           [string]
-      --useSitemap                          If enabled, check for sitemaps at
+      --useSitemap, --sitemap               If enabled, check for sitemaps at
```
Comment (emmadickson) [inline README.md:127]: do we want to aslo allow usesitemap since we do that for some other flags

File: util/argParser.js [hunk 2]
```diff
@@ -169,6 +170,7 @@ class ArgParser {
       },
 
       "useSitemap": {
+        alias: "sitemap",
```
Comment (emmadickson) [inline util/argParser.js:173]: might want to allow usesitemap

**Review by emmadickson — APPROVED** [review]
LGTM we'll need to change the tests for the pr I just pushed to

File: README.md (hunk 1 above)
Comment (ikreymer) [reply README.md:127]: Eh, maybe better to avoid so many variations?

**Pull request merged**